### ISSUE TITLE
[Source-postgres] : Compare each record’s lsn_commit value instead of lsn_proc. 

### ISSUE THREAD

**Issue by akashkulk** [airbytehq/airbyte#35963]
TLDR is we are comparing the lsn_proc from the event instead of the lsn_commit

If lsn_proc(record) > lsn_commit(state_file) AND lsn_commit(record) < lsn_comit(state_file) AND we emit state + fail
Can explain certain data loss situations, since we emit state that corresponds to records we may not have processed yet. We also ack this LSN and clear the WAL, so these records are forever lost. 

References : 
- https://docs.google.com/document/d/1yVW3QJjOZiCTHU4m-P-FdIm5yiJLzmomrFol9_7oPxc/edit
- https://debezium.zulipchat.com/#narrow/stream/348249-community-postgresql/topic/Different.20LSN.20values.20in.20Debezium.20state

**Comment by akashkulk** [comment]
Closed by https://github.com/airbytehq/airbyte/pull/35904
